csrc/device_lower/pass/allocation.cpp

```diff
@@ -661,6 +659,9 @@ class AllocationInserter : public kir::ExprMutator {
 
     if (circular_buffer_load_is_tma) {
       for (auto tv : circular_buffer_tvs) {
+        if (!ir_utils::isCpAsyncBulkLoad(tv->definition())) {
+          continue;
+        }
```

Comment:
This fixes a bug introduced in https://github.com/NVIDIA/Fuser/pull/3208, where we are creating mbarriers for circular-buffered non-TMA tensors.